### PR TITLE
UIDATIMP-727: Cover <TreeLine> component with tests

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose

Cover `<TreeLine>` component with tests using RTL+Jest.

## Approach

Covered `<TreeLine>` component with tests using RTL+Jest.

## Refs

https://issues.folio.org/browse/UIDATIMP-727

## Screenshots

image

### CODE REVIEW

File: src/components/TreeLine/TreeLine.test.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    renderTreeLine(treeLineWithContainer);
+    const line = container.children[0].children.length;
+
+    expect(line).toBeGreaterThan(2);
+  });
+
+  it('should be rendered inside document.body', () => {
+    const { container } = renderTreeLine(treeLineWithoutContainer);
+    const line = container.children;
```
Comment (mariia-aloshyna) [inline src/components/TreeLine/TreeLine.test.js:109]: This is a very unreliable test, any element can be children, it is not guaranteed that this is a line.

File: src/components/TreeLine/TreeLine.test.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    expect(line).toBeDefined();
+  });
+
+  it('should be rendered vertically', () => {
+    const { container } = render(parentContainer);
+
+    renderTreeLine(treeLineVertical);
+    const line = container.children[0].children.length;
+
+    expect(line).toBeGreaterThan(2);
```
Comment (mariia-aloshyna) [inline src/components/TreeLine/TreeLine.test.js:120]: This test does not check actually `vertical` line, children can be whatever.

File: src/components/TreeLine/TreeLine.test.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  });
+
+  it('should be rendered horizontally', () => {
+    const { container } = render(parentContainer);
+
+    renderTreeLine(treeLineHorizontal);
+    const line = container.children[0].children.length;
+
+    expect(line).toBeGreaterThan(2);
```
Comment (mariia-aloshyna) [inline src/components/TreeLine/TreeLine.test.js:129]: Here is the same

File: src/components/TreeLine/TreeLine.test.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  });
+
+  it('should be rendered one line', () => {
+    const { container } = render(parentContainer);
+
+    renderTreeLine(oneTreeLine);
+    const line = container.children[0].children.length;
+
+    expect(line).toBeGreaterThan(2);
```
Comment (mariia-aloshyna) [inline src/components/TreeLine/TreeLine.test.js:138]: And this one as well. All 3 tests look the same but should check different cases.

File: src/utils/resizeObserver.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export default class ResizeObserver {
+  observe() {
+    // do nothing
+  }
+
+  unobserve() {
+    // do nothing
+  }
+}
```
Comment (mariia-aloshyna) [inline src/utils/resizeObserver.js:9]: Remove this code, please, and create a mock for the `ResizeObserver` object inside your test

**Review by IvanKryzhanovskyi — APPROVED** [review]

File: src/components/TreeLine/TreeLine.test.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+};
+const treeLineWithoutContainer = {
+  from: '[data-id=HOLDINGS]',
+  to: '[data-id=INSTANCE]',
+  container: null,
+  fromAnchor: '12px 12px',
+  toAnchor: '13px 13px',
+  className: 'test',
+};
+const renderTreeLine = ({
```
Comment (mariia-aloshyna) [inline src/components/TreeLine/TreeLine.test.js:72]:
```suggestion
};

const renderTreeLine = ({
```

File: src/components/TreeLine/TreeLine.test.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+  return render(component);
+};
+
+window.ResizeObserver = jest.fn();
+window.ResizeObserver.mockImplementation(() => {
+  return {
+    observe() {},
+    unobserve() {},
+  };
+});
```
Comment (mariia-aloshyna) [inline src/components/TreeLine/TreeLine.test.js:108]: It's possible to write implementation inside `jest.fn()` as far as I remember, could you double-check it, please, and refactor if it is?
```suggestion
window.ResizeObserver = jest.fn(() => ({
    observe() {},
    unobserve() {},
}));
```

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**